src/plugins/mvcc_delete_plugin.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * This function processes the physical-delete-queue until its empty.
+ */
+void MvccDeletePlugin::_physical_delete_loop() {
+  std::unique_lock<std::mutex> lock(_mutex_physical_delete_queue);
+
+  ChunkSpecifier chunk_spec = _physical_delete_queue.front();
+  const auto& table = StorageManager::get().get_table(chunk_spec.table_name);
+  const auto& chunk = table->get_chunk(chunk_spec.chunk_id);
+
+  DebugAssert(chunk != nullptr, "Chunk does not exist. Physical Delete can not be applied.");
+
+  // Check whether there are still active transactions that might use the chunk
+  CommitID cleanup_commit_id = chunk->get_cleanup_commit_id();
+  CommitID lowest_snapshot_commit_id = TransactionManager::get().get_lowest_active_snapshot_commit_id();
+
+  if (cleanup_commit_id < lowest_snapshot_commit_id) {
+    _delete_chunk_physically(chunk_spec.table_name, chunk_spec.chunk_id);
+    _physical_delete_queue.pop();
+  }
+}
+
+bool MvccDeletePlugin::_delete_chunk_logically(const std::string& table_name, const ChunkID chunk_id) {
+  const auto& table = StorageManager::get().get_table(table_name);
+  const auto& chunk = table->get_chunk(chunk_id);
+
+  DebugAssert(chunk != nullptr, "Chunk does not exist. Physical Delete can not be applied.");
+  DebugAssert(chunk_id < (table->chunk_count() - 1),
+              "MVCC Logical Delete should not be applied on the last/current mutable chunk.");
+
+  // Create temporary referencing table that contains the given chunk only
+  auto transaction_context = TransactionManager::get().new_transaction_context();
+  auto gt = std::make_shared<GetTable>(table_name);
+  gt->set_transaction_context(transaction_context);
+
+  std::vector<ChunkID> excluded_chunk_ids(table->chunk_count());
+  std::iota(std::begin(excluded_chunk_ids), std::end(excluded_chunk_ids), 0);
+  excluded_chunk_ids.erase(std::remove(std::begin(excluded_chunk_ids), std::end(excluded_chunk_ids), chunk_id));
+
+  gt->set_excluded_chunk_ids(excluded_chunk_ids);
+  gt->execute();
+
+  // Validate temporary table
+  auto validate_table = std::make_shared<Validate>(gt);
+  validate_table->set_transaction_context(transaction_context);
+  validate_table->execute();
+
+  // Use Update operator to delete and re-insert valid records in chunk
+  // Pass validate_table into Update operator twice since data will not be changed.
+  auto update_table = std::make_shared<Update>(table_name, validate_table, validate_table);
+  update_table->set_transaction_context(transaction_context);
+  update_table->execute();
+
+  // Check for success
+  if (update_table->execute_failed()) {
+    transaction_context->rollback();
+    return false;
+  } else {
+    // TODO(all): Check for success of commit, currently (2019-01-11) not yet possible.
+    transaction_context->commit();
+
+    // Mark chunk as logically deleted
+    chunk->set_cleanup_commit_id(transaction_context->commit_id());
+    std::cout << "Deleted chunk " << chunk_id << " logically." << std::endl;
+    return true;
```

Comment:
You updated delete_chunk_physically, but this one still returns a bool. At least add "try"

Reply:
Makes sense, will be addressed.